src/fundus/scraping/crawler.py

```diff
@@ -30,17 +32,23 @@
 import requests
 from dateutil.rrule import MONTHLY, rrule
 from tqdm import tqdm
-from typing_extensions import ParamSpec
+from typing_extensions import ParamSpec, TypeAlias
 
-from fundus.publishers.base_objects import PublisherEnum
+from fundus.logging import basic_logger
+from fundus.publishers.base_objects import PublisherCollectionMeta, PublisherEnum
 from fundus.scraping.article import Article
-from fundus.scraping.common_crawl.html import CCNewsSource
-from fundus.scraping.common_crawl.scraper import CCNewsScraper
-from fundus.scraping.filter import ExtractionFilter, Requires, URLFilter
+from fundus.scraping.delay import Delay
+from fundus.scraping.filter import ExtractionFilter, Requires, RequiresAll, URLFilter
+from fundus.scraping.html import CCNewsSource
+from fundus.scraping.scraper import CCNewsScraper, WebScraper
+from fundus.scraping.session import session_handler
+from fundus.scraping.url import URLSource
 
 _T = TypeVar("_T")
 _P = ParamSpec("_P")
 
+Publisher: TypeAlias = Union[PublisherEnum, Type[PublisherEnum], PublisherCollectionMeta]
+
```

Comment:
Yes, all of those are different types and also valid. We use `more_itertools.collapse` so that's why it doesn't make a difference for the crawler in the end.